test/spec/services/simulationService.js

```diff
@@ -56,16 +15,20 @@ describe('SimulationService', function () {
             });
         });
 
-        it('should return the complement from droitsDescription with the given droits', function() {
-            var actual = service.complement({ acs: { montant: 12 } });
-
-            expect(actual.acs).toBe(undefined);
-            expect(actual.apl.id).toBe('apl');
+        it('should sort eligible and non eligible droits', function() {
+            var openfiscaResult = { acs: 10 };
+            var droits = service.sortDroits(openfiscaResult);
+            expect(droits.droitsEligibles).toEqual({ acs: { id: 'acs', montant: 10 } });
+            expect(droits.droitsNonEligibles).toEqual({ apl: { id: 'apl' } });
+            expect(droits.droitsInjectees).toEqual({});
```

Comment:
Yes, it would be better because the generated messages would be clearer.

If, for example, the algorithm that computes `droitsNonEligibles` fails, it is much clearer to have two messages that end with “…should compute droits non eligibles” under two different `describe` rather than knowing that both “should sort eligible and non eligible droits” and “should sort eligible and injected droits” fail. In the latter case, you have to find the lines and read the code to pinpoint the failure, while it is obvious otherwise.

You can also notice that the title of your tests is misleading: it states “it('should sort eligible and non eligible droits'”, yet you test `expect(droits.droitsInjectees).toEqual({})`. If that last line fails, the message will be `SimulationService sortDroits should sort eligible and non eligible droits failed: expected {a:1} to equal {}`, while that very sorting of eligible and non eligible will be fine.